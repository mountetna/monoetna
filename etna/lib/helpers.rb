require_relative './etna/clients'
require_relative './etna/environment_scoped'

module WithEtnaClients
  def environment
    EtnaApp.instance.environment
  end

  def token
    if !ENV['TOKEN']
      puts "No environment variable TOKEN is set. Set your token with `export TOKEN=<your.janus.token>` before running Etna commands."
      exit
    end
    ENV['TOKEN']
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(token: token, host: EtnaApp.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(token: token, host: EtnaApp.instance.config(:metis, environment)[:host])
  end

  def janus_client
    @janus_client ||= Etna::Clients::Janus.new(token: token, host: EtnaApp.instance.config(:janus, environment)[:host])
  end
end

module WithLogger
  def logger
    EtnaApp.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
