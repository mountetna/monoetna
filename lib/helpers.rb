require_relative './etna/clients'
require_relative './etna/environment_scoped'

module WithEtnaClients
  def environment
    EtnaApp.instance.environment
  end

  def exit(status=true)
    WithEtnaClients.exit(status)
  end

  # Abstraction used to prevent accidental exist in specs.
  def self.exit(status)
    Kernel.exit(status)
  end

  def token(ignore_environment: false)
    unless ignore_environment
      if environment == :many
        raise "You have multiple environments configured, please specify your environment by adding --environment <staging|production|development>"
      elsif environment == :none
        raise "You do not have a successfully configured environment, please run #{program_name} config set https://polyphemus.ucsf.edu"
      end
    end

    env_token = ENV['TOKEN']
    if !env_token
      puts "No environment variable TOKEN is set.  You should set your token with `export TOKEN=<your.janus.token>` before running."
      redirect = EtnaApp.instance.config(:auth_redirect)

      if redirect.nil? && EtnaApp.instance.environment == :production
        redirect = 'https://janus.ucsf.edu/'
      end

      unless redirect.nil?
        puts "Open your browser to #{redirect} to complete login and copy your token."
      end

      exit
    end

    env_token
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:magma, environment) || {})
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:metis, environment) || {})
  end

  def janus_client
    @janus_client ||= Etna::Clients::Janus.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:janus, environment) || {})
  end

  def polyphemus_client
    @polyphemus_client ||= Etna::Clients::Polyphemus.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:polyphemus, environment) || {})
  end
end

module WithLogger
  def logger
    EtnaApp.instance.logger
  end
end

module StrongConfirmation
  def confirm
    puts "Confirm Y/n:"
    input = STDIN.gets.chomp
    if input.downcase != "y"
      puts "Bailing..."
      return false
    end

    true
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
