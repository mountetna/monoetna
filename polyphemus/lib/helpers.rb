require 'etna/clients'
require 'etna/environment_scoped'
require 'shellwords'

module WithSlackNotifications
  def notify_slack(message, messenger:self.class.name, channel:)
    `/bin/post-to-slack.sh #{Shellwords.escape(messenger)} #{Shellwords.escape(channel)} #{Shellwords.escape(message)} || true`
  end
end

module WithEtnaClients
  def environment
    Polyphemus.instance.environment
  end

  def token
    Polyphemus.instance.config(:polyphemus, environment)[:token]
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
      token: token,
      host: Polyphemus.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
      token: token,
      host: Polyphemus.instance.config(:metis, environment)[:host])
  end

  def metis_client=(client)
    @metis_client = client if Polyphemus.instance.test?
  end
end

module WithLogger
  def logger
    Polyphemus.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
