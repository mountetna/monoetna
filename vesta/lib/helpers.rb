require 'etna/clients'
require 'etna/environment_scoped'
require 'shellwords'

module WithEtnaClients
  def environment
    Vesta.instance.environment
  end

  def token
    Vesta.instance.config(:vesta, environment)[:token]
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
      token: token,
      host: Vesta.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
      token: token,
      host: Vesta.instance.config(:metis, environment)[:host])
  end

  def metis_client=(client)
    @metis_client = client if Vesta.instance.test?
  end

  def janus_client
    @magma_client ||= Etna::Clients::Janus.new(
      token: token,
      host: Vesta.instance.config(:janus, environment)[:host])
  end
end

module WithLogger
  def logger
    Vesta.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
