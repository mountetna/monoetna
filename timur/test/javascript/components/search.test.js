import React from 'react';
import {Provider} from 'react-redux';
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {mockStore, stubUrl, cleanStubs} from '../helpers';
import Search, {
  Search as RawSearchComponent
} from '../../../lib/client/jsx/components/search/search';
import SelectInput from '../../../lib/client/jsx/components/inputs/select_input';

const models = {
  monster: {template: require('../fixtures/template_monster.json')},
  labor: {template: require('../fixtures/template_labor.json')},
  project: {template: require('../fixtures/template_project.json')}
};

describe('Search', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      magma: {models},
      search: {
        model_name: 'Labors',
        pages: {}
      }
    });
  });

  afterEach(cleanStubs);

  it('renders', () => {
    global.TIMUR_CONFIG = {
      magma_host: 'magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <Search />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('selecting a table (model) enables the Search and TSV buttons', async () => {
    const component = mount(
      <Provider store={store}>
        <Search />
      </Provider>
    );

    expect(component.find('.button.disabled').length).toEqual(2);

    const tableSelect = component.find(SelectInput).first();
    tableSelect.simulate('change', {
      target: {
        value: '0'
      }
    });

    // Trigger a re-render so that the buttons are updated
    component.update();

    expect(component.find('.button.disabled').length).toEqual(0);
  });

  it('resets the displayed attributes when a new table (model) is selected', () => {
    const component = mount(
      <Provider store={store}>
        <Search />
      </Provider>
    );

    const tableSelect = component.find(SelectInput).first();
    tableSelect.simulate('change', {
      target: {
        value: '0'
      }
    });

    // Trigger a re-render so that state and props are updated
    component.update();

    const first_attrs = component.find('.etna-checkbox');

    tableSelect.simulate('change', {
      target: {
        value: '1'
      }
    });

    component.update();

    const second_attrs = component.find('.etna-checkbox');
    expect(first_attrs.length).not.toEqual(second_attrs.length);
  });

  it('resets the cache when a new table (model) is selected', () => {
    const mockEmptySearchCache = jest.fn();

    const component = shallow(
      <RawSearchComponent
        magma_state={{models}}
        cache={{}}
        emptySearchCache={mockEmptySearchCache}
        requestModels={jest.fn()}
        setSearchAttributeNames={jest.fn()}
      />
    );

    expect(mockEmptySearchCache).toHaveBeenCalled();

    const tableSelect = component.find(SelectInput).first();
    // Different for shallow and mount?
    tableSelect.simulate('change', 'monster');

    component.update();

    tableSelect.simulate('change', 'labor');

    component.update();

    // One time was on mounting
    expect(mockEmptySearchCache).toHaveBeenCalled();
  });

  it('returns no attribute_names if no model selected', () => {
    const component = shallow(
      <RawSearchComponent
        magma_state={{models}}
        cache={{}}
        emptySearchCache={jest.fn()}
        requestModels={jest.fn()}
        setSearchAttributeNames={jest.fn()}
      />
    );

    expect(component.instance().getDisplayAttributeOptions()).toEqual(
      undefined
    );
  });

  it('nests attribute_names in arrays', () => {
    const component = shallow(
      <RawSearchComponent
        magma_state={{models}}
        cache={{}}
        emptySearchCache={jest.fn()}
        requestModels={jest.fn()}
        setSearchAttributeNames={jest.fn()}
      />
    );

    const tableSelect = component.find(SelectInput).first();
    // Different for shallow and mount?
    tableSelect.simulate('change', 'monster');

    component.update();

    expect(component.instance().getDisplayAttributeOptions()).toEqual([
      ['aspect'],
      ['labor'],
      ['name'],
      ['species'],
      ['stats'],
      ['victim']
    ]);
  });

  it('calculates tree state with selected and unselected attribute names', () => {
    const component = shallow(
      <RawSearchComponent
        magma_state={{models}}
        cache={{}}
        emptySearchCache={jest.fn()}
        requestModels={jest.fn()}
        setSearchAttributeNames={jest.fn()}
      />
    );

    const result = component
      .instance()
      .convertAttributeNameListToTreeState(
        ['labor', 'species', 'stats'],
        [['aspect'], ['labor'], ['name'], ['species'], ['stats'], ['victim']]
      );

    expect(result).toEqual({
      aspect: false,
      labor: true,
      name: false,
      species: true,
      stats: true,
      victim: false
    });
  });
});
