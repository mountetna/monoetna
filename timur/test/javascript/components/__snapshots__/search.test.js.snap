// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<div
  id="search"
>
  <div
    className="control"
  >
    <div
      className="query"
    >
      <span
        className="label"
      >
        Show table
      </span>
      <select
        defaultValue=""
        onChange={[Function]}
      >
        <option
          disabled={false}
          value=""
        >
           --- 
        </option>
        <option
          value={0}
        >
          labor
        </option>
        <option
          value={1}
        >
          monster
        </option>
        <option
          value={2}
        >
          project
        </option>
      </select>
      <span
        className="label"
      >
        Page size
      </span>
      <select
        defaultValue={10}
        onChange={[Function]}
      >
        <option
          disabled={true}
          value=""
        >
           --- 
        </option>
        <option
          value={0}
        >
          10
        </option>
        <option
          value={1}
        >
          25
        </option>
        <option
          value={2}
        >
          50
        </option>
        <option
          value={3}
        >
          200
        </option>
      </select>
      <input
        className="filter"
        onBlur={[Function]}
        placeholder="Filter query"
        type="text"
      />
      <input
        className="button disabled"
        disabled={true}
        id="search-pg-search-btn"
        onClick={[Function]}
        type="button"
        value="Search"
      />
      <input
        className="button disabled"
        disabled={true}
        id="search-pg-tsv-btn"
        onClick={[Function]}
        type="button"
        value="â†¯ TSV"
      />
    </div>
  </div>
  <div
    className="documents"
  >
    <div
      className="table"
    />
  </div>
</div>
`;
